broker/patron_request/service/ncip_adapter.go

```diff
@@ -0,0 +1,58 @@
+package prservice
+
+import (
+	"github.com/indexdata/crosslink/broker/adapter"
+	"github.com/indexdata/crosslink/broker/ncipclient"
+	"github.com/indexdata/crosslink/ncip"
+)
+
+type NcipAdapter struct {
+	ncipClient ncipclient.NcipClient
+	dirAdapter adapter.DirectoryLookupAdapter
+}
+
+func CreateNcipAdapter(ncipClient ncipclient.NcipClient, dirAdapter adapter.DirectoryLookupAdapter) NcipAdapter {
+	return NcipAdapter{
+		ncipClient: ncipClient,
+		dirAdapter: dirAdapter,
+	}
+}
+
+func (n *NcipAdapter) getNcipInfoFromSymbols(symbols []string) (map[string]any, error) {
+	dirEntries, _, err := n.dirAdapter.Lookup(adapter.DirectoryLookupParams{
+		Symbols: symbols,
+	})
+	if err != nil {
+		return nil, err
+	}
+	// find first NCIP entry
+	for _, entry := range dirEntries {
+		customData := entry.CustomData
+		if ncip, ok := customData["ncip"].(map[string]any); ok {
+			return ncip, nil
+		}
+	}
+	return nil, nil
+}
+
+func (n *NcipAdapter) LookupUser(symbols []string, user string, password string) (bool, error) {
```

Comment:
I think this (and other NCIP functions) function should take `symbol` string scalar (going forward `Peer` entity) and return the NCIP response, not boolean. We will need the responses later when we check patron blocks, etc.